pygmt/src/grdproject.py

```diff
@@ -10,13 +10,14 @@
 from pygmt.alias import Alias, AliasSystem
 from pygmt.clib import Session
 from pygmt.exceptions import GMTInvalidInput
-from pygmt.helpers import build_arg_list, fmt_docstring, use_alias
+from pygmt.helpers import build_arg_list, deprecate_parameter, fmt_docstring, use_alias
 
 __doctest_skip__ = ["grdproject"]
 
 
 @fmt_docstring
-@use_alias(E="dpi", F="scaling", I="inverse", M="unit", n="interpolation")
+@deprecate_parameter("unit", "units", "v0.18.0", remove_version="v0.20.0")
+@use_alias(E="dpi", F="scaling", I="inverse", M="units", n="interpolation")
```

Comment:
I think it should be `unit`, not `units`, since it takes one single unit in this module.

Reply:
I was wondering it it makes sense to have this consistent across function / methods and always use plural.